### ISSUE TITLE
Code Cleanup

### ISSUE THREAD

**Issue by JasonLG1979** [JasonLG1979/gnome-shell-extensions-mediaplayer#389]
The code in the extension is a mess, which is 100% my fault. I try my best to make sure it doesn't throw any unhandled exceptions or errors but beyond that I am not a js programmer and have, at the time of this writing about 6 months worth of js noodling under my belt. I'd like some help cleaning up the style and organization of the code as phase 1.  As phase 2 I'd like help optimizing the code a bit. I have a rough grasp and some experience with the Gtk/GLib/Gio bits but the Gjs/St.Widget/Shell Widgets docs leave a lot to be desired. The stock widgets in particular suck for use with this extension's UI. I do a lot wrestling with the widgets to get them to behave in a way that I'd like. I'd like to minimize that as much as possible.

**Comment by lexruee** [comment]
@JasonLG1979 
First of all, thanks for this awesome extension!

Just found this issue here. I'm not a pro JS programmer but a python programmer. Nevertheless, I contributed [some code](https://github.com/LinxGem33/Arc-Menu/graphs/contributors) to Arc-Menu and [two fixes to GNOME](https://bugzilla.gnome.org/page.cgi?id=describeuser.html&login=a.rueedlinger%40gmail.com). So I have gained some experience in Gjs/GTk+/St stuff. However, what I like is to improve and refactor code as a way to learn more about gnome shell development :smile:.

For example, I could work together with @LinxGem33 to prettify the existing preferences UI/UX.
Here is a screenshot what we [Arc-Menu Team] have implemented for the Arc-Menu preferences:
![screenshot from 2017-07-26 01-42-36](https://user-images.githubusercontent.com/649340/28598419-c947dd40-71a3-11e7-999d-c390b84d9a70.png)

Let me or us know what you think about this suggestion.

PS: I like your code of conduct :-).

**Comment by JasonLG1979** [comment]
> First of all, thanks for this awesome extension!

Thanks. I'm not the original author, I just took over the project 99.999% of the credit should go to @eonpatapon.

> Let me know what you think about this suggestion.

I'm for sure not against improving the preferences dialog. I've been meaning to at the very least make it scroll able.

> PS: I like your code of conduct :-).

Thanks. I've read a lot of codes of conducts for different projects and they all just basically boil down to "Don't be an A-hole." But they take 75 paragraphs to say that,lol!!!

**Comment by JasonLG1979** [comment]
@lexruee still waiting on that Prefs UI redesign ;P

**Comment by JasonLG1979** [comment]
> Hi, good things need time :-). @LinxGem33 and I are currenlty planning the next release of Arc-Menu :-).

Right on.

**Comment by JasonLG1979** [comment]
@lexruee This covers just about what I wanted to do:

https://github.com/JasonLG1979/gnome-shell-extensions-mediaplayer/commit/984f398b0622768257c28ba28c22416bdfa4f291

I got a lot of inspiration from the prefs in ArcMenu.

**Comment by lexruee** [comment]
@JasonLG1979 
Nice job :smile:. It looks great!

But I would also add an About page with a nice logo and a link to your github repo. In this case you could remove the "Visit the Github page..." link at the bottom of the prefs widget.

**Comment by JasonLG1979** [comment]
It still needs a little work. An About page would be nice. Still haven't settled on a logo so we can skip that part for now. It would be nice if the about page otherwise mimics the stock Gtk about dialog.

**Comment by JasonLG1979** [comment]
Another issue I'd like to fix is that the notebook widget does not expand to fill the window when the window is resized. It would be better if it did.

**Comment by lexruee** [comment]
@JasonLG1979 
The about page looks nice. Well done. I hope you don't mind if I cherry-pick some parts for Arc-menu :-).

**Comment by JasonLG1979** [comment]
By all means use what you like. After all I borrowed heavily from Arc-Menu's prefs. Share and share a like ;)

**Comment by JasonLG1979** [comment]
I did my best to keep the "factory" aspect of the prefs so anyone can easily add an option without having to rewrite the whole thing.

**Comment by JasonLG1979** [comment]
https://github.com/JasonLG1979/gnome-shell-extensions-mediaplayer/blob/master/README.md
